readme.md

```diff
@@ -246,123 +240,157 @@ Defaults to `'Too many requests, please try again later.'`
+Defaults to `false` (for backward compatibility, but its use is recommended).
 
 > Renamed in `6.x` from `draft_polli_ratelimit_headers` to `standardHeaders`.
 
+### `requestPropertyName`
+
+> `string`
+
+The name of the property on the Express `request` object to store the rate limit
+info.
+
+Defaults to `'rateLimit'`.
+
+### `skipFailedRequests`
+
+> `boolean`
+
+When set to `true`, failed requests won't be counted. Request considered failed
+when the `requestWasSuccessful` option returns `false`. By default, this means
+requests fail when:
+
+- the response status >= 400
+- the request was cancelled before last chunk of data was sent (response `close`
+  event triggered)
+- the response `error` event was triggered by response
+
+(Technically they are counted and then un-counted, so a large number of slow
+requests all at once could still trigger a rate-limit. This may be fixed in a
+future release. PRs welcome!)
+
+Defaults to `false`.
+
+### `skipSuccessfulRequests`
+
+> `boolean`
+
+If `true`, the library will (by default) skip all requests that are considered
+'failed' by the `requestWasSuccessful` function. By default, this means requests
+succeed when the response status code < 400.
+
+(Technically they are counted and then un-counted, so a large number of slow
+requests all at once could still trigger a rate-limit. This may be fixed in a
+future release. PRs welcome!)
+
+Defaults to `false`.
+
 ### `keyGenerator`
 
-Function used to generate keys.
+> `function`
 
-Defaults to `request.ip`, similar to this:
+Method to generate custom identifiers for clients.
+
+By default, the client's IP address is used:
 
 ```ts
-const keyGenerator = (request /*, response*/) => request.ip
+const keyGenerator = (request, response) => request.ip
 ```
 
 ### `handler`
 
-The function to handle requests once the max limit is exceeded. It receives the
-`request` and the `response` objects. The `next` param is available if you need
-to pass to the next middleware/route. Finally, the `options` param has all of
-the options that originally passed in when creating the current limiter and the
-default values for other options.
+> `function`
 
-The `request.rateLimit` object has `limit`, `current`, and `remaining` number of
-requests and, if the store provides it, a `resetTime` Date object.
+Express request handler that sends back a response when a client is
+rate-limited.
 
-Defaults to:
+By default, sends back the `statusCode` and `message` set via the options:
 
 ```ts
 const handler = (request, response, next, options) => {
 	response.status(options.statusCode).send(options.message)
 }
 ```
 
-### `requestWasSuccessful`
+### `onLimitReached`
 
-Function that is called when `skipFailedRequests` and/or
-`skipSuccessfulRequests` are set to `true`. May be overridden if, for example, a
-service sends out a 200 status code on errors.
+> `function`
 
-Defaults to
+Express request handler that sends back a response when a client has reached
```

Comment:
I don't like calling it a request handler, because that really wasn't the intention (and tends to lead to people confusing it with the `handler` option). 

Also, I think it happens on the first limited request, not the one before the it.

But, meh, it's deprecated because it was so confusing....